src/views/AddItem.jsx

```diff
@@ -1,39 +1,80 @@
 import { useEffect, useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listId }) {
+export function AddItem({ data, listId }) {
 	const [timeframe, setTimeframe] = useState('7');
 	const [itemName, setItemName] = useState('');
-	const [message, setMessage] = useState(null);
+	const [messages, setMessages] = useState([]);
 
 	useEffect(() => {
 		const timer = setTimeout(() => {
-			setMessage(null);
+			setMessages([]);
 		}, 3000);
 
 		return () => clearTimeout(timer);
-	}, [message]);
+	}, [messages]);
 
 	const onTimeChange = (e) => setTimeframe(e.target.value);
 	const onItemChange = (e) => setItemName(e.target.value);
+
+	// function to validate itemName input that returns error messages if any - called onFormSubmit
+	function validateItemInput(data, trimmedItemName) {
+		// 1- checks for empty inputs
+
+		if (!trimmedItemName) {
+			return ['Please enter an item name.'];
+		}
+
+		const errArray = [];
+
+		// 2- checks for invalid characters
+
+		if (trimmedItemName.search(/[^a-z0-9'&-\s]/i) !== -1) {
+			errArray.push(
+				`Please enter an item name that is alphanumeric or includes ', -, and &.`,
+			);
+		}
+
+		const normalizedItemName = trimmedItemName
+			.replace(/[^a-z0-9'&-]/gi, '')
+			.toLowerCase();
+
+		const potentialMatch = data.find(
+			(item) =>
+				item.name.replaceAll(' ', '').toLowerCase() === normalizedItemName,
+		);
+
+		// 3- checks for potential match including exact match and match after removing special characters and spaces
+
+		if (potentialMatch) {
+			errArray.push(
+				`${trimmedItemName} is already on your list as ${potentialMatch.name}.`,
+			);
+		}
+
```

Comment:
Good implementation. It's thorough and it handles all the edge cases discussed in Slack!